File: src/main/java/me/centralhardware/znatoki/telegram/statistic/service/TelegramService.java
package me.centralhardware.znatoki.telegram.statistic.service;

import lombok.RequiredArgsConstructor;
import me.centralhardware.znatoki.telegram.statistic.entity.postgres.Role;
import me.centralhardware.znatoki.telegram.statistic.entity.postgres.TelegramUser;
import me.centralhardware.znatoki.telegram.statistic.mapper.postgres.UserMapper;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class TelegramService {

    private final UserMapper userMapper;

    public boolean hasWriteRight(Long chatId) {
        var role = getRole(chatId);
        return role == Role.READ_WRITE ||
                role == Role.ADMIN;
    }

    public boolean hasReadRight(Long chatId) {
        var role = getRole(chatId);
        return role == Role.READ ||
                role == Role.READ_WRITE ||
                role == Role.ADMIN;
    }

    public boolean isAdmin(Long chatId) {
        return getRole(chatId) == Role.ADMIN;
    }

    public boolean isUnauthorized(Long chatId) {
        return getRole(chatId) == null;
    }

    private Role getRole(Long chatId){
        return Optional.ofNullable(userMapper.getById(chatId))
                .map(TelegramUser::getRole)
                .orElse(null);
    }

}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/service/TelegramService.kt
package me.centralhardware.znatoki.telegram.statistic.service

import me.centralhardware.znatoki.telegram.statistic.entity.Role
import me.centralhardware.znatoki.telegram.statistic.mapper.UserMapper
import org.springframework.stereotype.Service

@Service
class TelegramService(private val userMapper: UserMapper) {

    fun hasWriteRight(chatId: Long): Boolean {
        val role = getRole(chatId)
        return role == Role.READ_WRITE ||
                role == Role.ADMIN
    }

    fun hasReadRight(chatId: Long): Boolean {
        val role = getRole(chatId)
        return role == Role.READ ||
                role == Role.READ_WRITE ||
                role == Role.ADMIN
    }

    fun isAdmin(chatId: Long): Boolean {
        return getRole(chatId) == Role.ADMIN
    }

    private fun getRole(chatId: Long): Role? {
        return userMapper.getById(chatId)?.role
    }

}